publisher-sdk/src/main/java/com/criteo/publisher/logging/Logger.java

```diff
@@ -33,6 +33,42 @@
   @NonNull
   private final List<LogHandler> handlers;
 
+  /**
```

Comment:
Such code can change, so I added the reason in the commit message rather than in the code directly.

Reply:
> Such code can change

Well, if the code changes then the comment would be removed altogether. I really think this is a low-hanging fruit improvement that can make a big difference in understanding the code base. Not everyone would think about reading the commit message to understand why this code exists in the first place. Think about ergonomics: we all like to be 1-click away from achieving something, same thing here: it is best to have the information readily available when reading the code.

Reply:
The issue with the comment in the code in that it will refer to something that is not next to the logger. So I really doubt that anyone will think about updating the comment in the logger class if a new `logger.something` is added/removed in another component.

For instance, there, you suggested to add a warning: https://github.com/criteo/android-publisher-sdk/pull/168#discussion_r538243292. This code is used by the remote logger, and adding a warning would make it responsible of such infinite recursion. By adding such warning, honestly, I would never imagine to update a comment sitting in the logger class.

So from my point of view, having a comment listing all cases involved in infinite recursion is not maintainable at all. At most, I can give you, now, what is the current state, and such snapshot belongs to a git message.

IMHO, reading the last version of  code is fine to learn what the code is doing, but to learn the why, you need to read the history (git blame) as well.

Reply:
The point was not to document every case that needs this hack, but to document the type of scenarios needed by it. For instance: 

```
Checking for recursive logs is necessary because logging can throw an exception (for example when performing an I/O operation to build a log message). Throwing an exception will result in an additional log message, which can throw again... hence resulting in an infinite loop.
```
Even if more code is added that could generate such issues, you don't have to update the comment as it is generic enough. That way, the reader understands the context in which this hack was written and is not left wondering why such thing exists. Also, commit message history can be rewritten/removed and this info can be lost.

Reply:
Oh ok, I didn't get that in the first place.

Sure, I will add this message

Reply:
Done

Reply:
Awesome, thanks